From: David Sterba <dsterba@suse.com>
Date: Thu, 8 Oct 2015 14:14:16 +0200
Patch-mainline: Not yet
References: bsc#949472
Subject: [PATCH] btrfs: change max_inline default to 2048

The current practical default is ~4k on x86_64 (the logic is more complex,
simplified for brevity), the inlined files land in the metadata group and
thus consume space that could be needed for the real metadata.

The inlining brings some usability surprises:

1) MgE measured total space consumption on various filesystems and btrfs
with DUP metadata was quite visible because of the duplicated data within
metadata.

2) inlined data may exhaust the metadata, which are more precious in case
when entire device space is allocated to chunks (ie. balance cannot to
make the space more compact)

3) performance suffers a bit as the inlined blocks are duplicate and
stored far away on the device.

Proposed fix: set the default to 2048

This fixes namely 1), the total filesysystem space consumption will be on
par with other filesystem.

Partially fixes 2), more data are pushed to the data block groups

3) different characteristic, but depends on the distribution of inlined
file sizes.

Signed-off-by: David Sterba <dsterba@suse.com>
---
 fs/btrfs/ctree.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/btrfs/ctree.h
+++ b/fs/btrfs/ctree.h
@@ -2084,7 +2084,7 @@ struct btrfs_ioctl_defrag_range_args {
 #define	BTRFS_MOUNT_CHANGE_INODE_CACHE	(1 << 24)
 
 #define BTRFS_DEFAULT_COMMIT_INTERVAL	(30)
-#define BTRFS_DEFAULT_MAX_INLINE	(8192)
+#define BTRFS_DEFAULT_MAX_INLINE	(2048)
 
 #define btrfs_clear_opt(o, opt)		((o) &= ~BTRFS_MOUNT_##opt)
 #define btrfs_set_opt(o, opt)		((o) |= BTRFS_MOUNT_##opt)
